server/db/db.js

```diff
@@ -325,6 +330,152 @@ module.exports.getInstallationWithInstallationId = async function(installationId
+    const values = [hub.flicLastSeenTime, hub.flicLastPingTime, hub.heartbeatLastSeenTime, hub.systemId]
+    await client.query(query, values) 
+    
+    if(!transactionMode) {
+        client.release()
+    }
+}
+
+module.exports.saveHubRename = async function(hub, client) {
+    let transactionMode = (typeof client !== 'undefined')
+    if(!transactionMode) {
+        client = await pool.connect()
+    }
+    
+    let { rows } = await client.query("SELECT * FROM hubs WHERE system_id = $1 LIMIT 1", [hub.systemId])
+    if(rows.length === 0) {
+        if(!transactionMode) {
+            client.release()
+        }
+        throw new Error("Tried to rename a hub that doesn't exist yet.")
+    }
+    const query = "UPDATE hubs SET system_name = $1 WHERE system_id = $2"
+    const values = [hub.systemName, hub.systemId]
+    await client.query(query, values) 
+    
+    if(!transactionMode) {
+        client.release()
+    }
+}
+
+module.exports.saveHubMuteStatus = async function(hub, client) {
+    let transactionMode = (typeof client !== 'undefined')
+    if(!transactionMode) {
+        client = await pool.connect()
+    }
+    
+    let { rows } = await client.query("SELECT * FROM hubs WHERE system_id = $1 LIMIT 1", [hub.systemId])
+    if(rows.length === 0) {
+        if(!transactionMode) {
+            client.release()
+        }
+        throw new Error("Tried to save mute status in a hub that doesn't exist yet.")
+    }
+    const query = "UPDATE hubs SET muted = $1 WHERE system_id = $2"
+    const values = [hub.muted, hub.systemId]
+    await client.query(query, values) 
+    
+    if(!transactionMode) {
+        client.release()
+    }
+}
+
+module.exports.saveHubHideStatus = async function(hub, client) {
+    let transactionMode = (typeof client !== 'undefined')
+    if(!transactionMode) {
+        client = await pool.connect()
+    }
+    
+    let { rows } = await client.query("SELECT * FROM hubs WHERE system_id = $1 LIMIT 1", [hub.systemId])
+    if(rows.length === 0) {
+        if(!transactionMode) {
+            client.release()
+        }
+        throw new Error("Tried to save hide status for a hub that doesn't exist yet.")
+    }
+    const query = "UPDATE hubs SET hidden = $1 WHERE system_id = $2"
+    const values = [hub.hidden, hub.systemId]
+    await client.query(query, values) 
+    
+    if(!transactionMode) {
+        client.release()
+    }
+}
+
+module.exports.saveHubAlertStatus = async function(hub, client) {
+    let transactionMode = (typeof client !== 'undefined')
+    if(!transactionMode) {
+        client = await pool.connect()
+    }
+    
+    let { rows } = await client.query("SELECT * FROM hubs WHERE system_id = $1 LIMIT 1", [hub.systemId])
+    if(rows.length === 0) {
+        if(!transactionMode) {
+            client.release()
+        }
+        throw new Error("Tried to save hide status for a hub that doesn't exist yet.")
```

Comment:
This comment should change `hide status` to `alert sent status` or something like that, right?